lazyflow/utility/reorderAxesDecorator.py

```diff
@@ -0,0 +1,447 @@
+                        self._inner_call = False
+                        return ret
+                    except Exception:
+                        raise
+                    finally:
+                        self._inner_call = False
+            return wrap
+
+        setattr(cls, meth, guard(old_meth))
+
+    # overwrite property methods with guarded versions thereof
+    for meth in property_methods:
+        old_get = getattr(cls, meth).__get__
+
+        def guard(fn):
+            @property
+            def wrap(self):
+                if self._inner_call:
+                    return fn(self)
+                else:
+                    try:
+                        self._inner_call = True
+                        ret = fn(self)
+                        self._inner_call = False
+                        return ret
+                    except Exception:
+                        raise
+                    finally:
+                        self._inner_call = False
+            return wrap
+
+        setattr(cls, meth, guard(old_get))
+
+    return cls
+
+
+def reorder(cls):
+    """
+        Decorator function to reorder all input and output channels, to preserve an inner axes order within the
+        operator.
+        Must be used in combination with reorder_options!!!
+
+        For every slot of a decorated operator, the distinction has to be made, if the slot was called by the operator
+        itself, i.e. by one of its methods, or on the operator object from the 'outside'.
+
+        Pseudo example to clarify:
+
+        @reorder
+        @reorder_options(internal_axis_order='set in stone by dev',
+                         exceptions: 'i_already_follow_the_new_outside_order',
+                         output_axes_order: 'the bright future')
+        class MyOp(operator):
+            Input = InputSlot()
+            Output = OutputSlot()
+
+            anarchySlot = InputSlot()
+
+            def __init__(self):
+                super().__init__()
+                self.childOp = ChildOp()
+
+                # direct connection of slots, not affected by reorder decorator
+                self.childOp.InputA.connect(self.anarchySlot)
+
+            def setupOutputs(self):
+                # everything in here is automatically converted with respect to the inner and outer axes orders
+                # In-between childOp's slot and this operator's slot a hidden OpReorderAxes is connected, which was
+                # added by the reorder decorator
+                self.childOp.InputB.connect(self.Input)
+                self.Output.connect(self.childOp.Output)
+    """
+    if '_reorder_exceptions' not in dir(cls):
+        cls._reorder_exceptions = []
+
+    cls = guard_methods(cls)
+
+    inputSlots = [s.name for s in cls.inputSlots
+                  if s.name not in cls._reorder_exceptions]
+    outputSlots = [s.name for s in cls.outputSlots
+                   if s.name not in cls._reorder_exceptions]
+
+    inner_prefix = 'inner_'
```

Comment:
not used?

Reply:
are you an archeologist? 'cause you just found an ancient artifact!